collector/manager/metrics_collector_manager.go

```diff
@@ -151,20 +151,32 @@ func (mcm *MetricsCollectorManager) declareMetricDefinitions(metricDetails []col
 
 	metricDefs := make([]hmetrics.MetricDefinition, len(endpoint.Metrics))
 
-	// For each metric in the endpoint, create a metric def for it.
-	// Notice metric tags override endpoint tags which override global tags.
-	// Do NOT allow pods to use agent environment variables since agent env vars may contain
-	// sensitive data (such as passwords). Only the global agent config can define tags
-	// with env var tokens.
-	globalTags := mcm.Config.Collector.Tags.ExpandTokens(true, additionalEnv)
-	endpointTags := endpoint.Tags.ExpandTokens(false, additionalEnv)
-
-	// we need these to expand tokens in the IDs
-	mappingFunc := expand.MappingFunc(false, additionalEnv)
-	mappingFuncWithEnv := expand.MappingFunc(true, additionalEnv)
-
 	for i, metric := range endpoint.Metrics {
 
+		// Define additional envvars with pod specific data for use in replacing ${env} tokens in tags.
+		env := map[string]string{
+			"METRIC:name":     	metric.Name,
+			"METRIC:id":	   	metric.ID,
+			"METRIC:units":	    	metric.Units,
+			"METRIC:description":	metric.Description,
```

Comment:
make sure to re-arrange this code with lines ~210-222 so that the actual description and units get placed here (which may or may not be what is in the metric.Units/metric.Description. Just have to move up what lines 210-222 do above here before putting the units/description in the env map.